install/install.go

```diff
@@ -675,12 +675,25 @@ func (k *K8sInstaller) generateAgentDaemonSet() *appsv1.DaemonSet {
 
 	}
 
-	mountCmd := `mount | grep "/sys/fs/bpf type bpf" || { echo "Mounting eBPF filesystem..."; mount bpffs /sys/fs/bpf -t bpf; }`
 	nodeInitContainers = append(nodeInitContainers, corev1.Container{
 		Name:            "ebpf-mount",
 		Image:           k.fqAgentImage(),
 		ImagePullPolicy: corev1.PullIfNotPresent,
-		Command:         []string{"nsenter", "--mount=/hostproc/1/ns/mnt", "--", "sh", "-c", mountCmd},
+		Command: []string{
+			"sh", "-ec",
+
+			// The statically linked Go program binary is invoked to avoid any
+			// dependency on utilities like sh and mount that can be missing on certain
+			// distros installed on the underlying host. Copy the binary to the
+			// same directory where we install cilium cni plugin so that exec permissions
+			// are available.
+			//
+			// See https://github.com/cilium/cilium/issues/17883
+			"cp /usr/bin/cilium-mount /hostbin/cilium-mount && " +
```

Comment:
The mountCmd that you are replacing at L678 is to mount the BPF filesystem. However, L692-696 are for mounting cgroup fs.

Reply:
Looks like we've already mounting BPF here:
https://github.com/cilium/cilium-cli/blob/d2d7498da6c1642115f98a74091063ab7eae5d59/install/install.go#L494-L498

I've changed `MountPropagation` to `bidirectional`, reflecting https://github.com/cilium/cilium/pull/16656.

So I've updated #L678-L745 also with some context from #627

PTAL, should be ok now.